Description: Enable junit4 use in Debian builds
Author: Olek Wojnar <olek@debian.org>
Forwarded: https://github.com/bazelbuild/bazel/pull/12755
Last-Update: 2020-12-27

--- a/third_party/BUILD
+++ b/third_party/BUILD
@@ -624,8 +624,9 @@
     jars = ["guava/guava-testlib-29.0-jre.jar"],
 )
 
-java_import(
+distrib_java_import(
     name = "junit4",
+    enable_distributions = ["debian"],
     jars = [
         "hamcrest/hamcrest-core-1.3.jar",
         "junit/junit-4.13.jar",
--- a/tools/distributions/debian/debian_java.BUILD
+++ b/tools/distributions/debian/debian_java.BUILD
@@ -449,3 +449,12 @@
         "grpc-stub.jar",
     ],
 )
+
+# junit4
+java_import(
+    name = "junit4",
+    jars = [
+        "hamcrest-core.jar",
+        "junit4.jar",
+    ],
+)
